ethers-solc/src/lib.rs

```diff
@@ -194,15 +197,22 @@ impl<Artifacts: ArtifactOutput> Project<Artifacts> {
             if !self.allowed_lib_paths.0.is_empty() {
                 solc = solc.arg("--allow-paths").arg(self.allowed_lib_paths.to_string());
             }
+            solc_versions.insert(solc.solc.clone(), version);
             let entry = sources_by_version.entry(solc).or_insert_with(BTreeMap::new);
-            entry.insert(path, source);
+            entry.insert(path.clone(), source);
         }
 
         let mut compiled =
             ProjectCompileOutput::with_ignored_errors(self.ignored_error_codes.clone());
 
         // run the compilation step for each version
         for (solc, sources) in sources_by_version {
+            // verify that this solc version's checksum matches the checksum found remotely
+            let version = solc_versions.get(&solc.solc).unwrap();
+            while let Err(_e) = solc.verify_checksum() {
+                Solc::blocking_install(version)?;
+            }
```

Comment:
what's the reasoning for the loop here, is this just to retry?

Reply:
Yea, its just to retry. The `blocking_install` would fail if the checksum didn't match on download. So it might actually be unnecessary. Would you suggest removing it?

I actually feel, in future if a bug creeps in, it will be extremely hard to debug this (as we would go into an infinite loop)